### PR TITLE
PLUGIN-1525 fix colon in password for sftp

### PR DESCRIPTION
**Pull request by albertshau**
fix the logic to extract the password to be a little better and work properly if the password contains a colon. This does not fix issues if the username contains a colon, and only fixes issues for SFTP and not FTP. This is because the underlying Hadoop FTPFileSystem also does not handle colons in username or password properly, while the SFTPFileSystem does not have that same bug.

### CODE REVIEW

File: src/test/java/io/cdap/plugin/batch/source/ftp/FTPBatchSourceTest.java [hunk 1]
```diff
@@ -82,8 +82,7 @@ public static void ftpTeardown() {
 
   @Test
   public void testSchemaDetection() {
-    String path = String.format("ftp://user2:%s@%s:%d/tmp/file1.txt",
-                                PASSWORD_WITH_SPECIAL_CHARACTERS, HOST, ftpServerPort);
+    String path = String.format("ftp://user:password@%s:%d/tmp/file1.txt", HOST, ftpServerPort);
```
Comment (albertshau) [inline src/test/java/io/cdap/plugin/batch/source/ftp/FTPBatchSourceTest.java:85]: had to change this because Hadoop's FTPFileSystem does not work if there is a `:` in the password (SFTP doesn't have this problem). The other tests check the properties are correct without attempting to actually read, so they can test that password is properly set.

**Review by tivv — APPROVED** [review]
Overall, LGTM
I am wondering if we should just use URI instead of `latIndexOf("@")` as there can be other corner cases. Also it would be nice to enhance validation piece of the plugin that user name does not have ":"

**Comment by albertshau** [comment]
it would definitely be better, but the existing code is bad and actually allows invalid URIs so I can't use it to extract it out.

I've opened a jira to improve this plugin, I don't think we should even have a path at all, but separate properties for host, port, path, username, and password. Once we do that all of this hacky parsing logic would be gone.
